README.md

```diff
@@ -1,115 +1,82 @@
 
-## Why use `stable_hash` instead of `Base.hash`?
+> ⚠️ Hash versions 3 constitutes a substantial redesign of StableHashTraits so as to avoid reliance on some unstable Julia internals. Hash versions 1 and 2 are deprecated and will be removed in a soon-to-be released StableHashTraits@2.0. Hash version 3 will remain unchanged in this 2.0 release. Hash version 1 is the default version if you don't specify a version.
 
-This package can be useful any time one of the following holds:
+<!--The START_ and STOP_ comments are used to extract content that is also repeated in the documentation-->
+<!--START_OVERVIEW-->
+## Use Case and Design Rationale
 
-- you want to ensure the hash value will not change when you update Julia or start a new session
-- you want to compute a hash for an object that does not have `hash` defined
-- you want to customize how the hash works, within a specific scope
+StableHashTraits is designed to be used in cases where there is an object you wish to serialize in a content-addressed cache. How and when objects pass the same input to a hashing algorithm is meant to be predictable and well defined, so that the user can reliably define methods of `transformer` to modify this behavior.
 
-This is useful for content-addressed caching, in which e.g. some function of a value is stored at a location determined by a hash. Given the value, one can recompute the hash to determine where to look to see if the function evaluation on that value has already been cached.
+## What gets hashed?
 
-## Details
+By default, an object is hashed according to its `StructType` (ala
+[SructTypes](https://github.com/JuliaData/StructTypes.jl)), and this can be customized using
+[`StableHashTraits.transformer`](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.transformer).
 
-You compute hashes using `stable_hash`. This is called on the object you want to hash, and
-(optionally) a second argument called the context. The context you use affects how hashing
-occurs (it defaults to `HashVersion{1}()`). It is generally recommended that you avoid
-`HashVersion{1}()`, favoring `HashVersion{2}()` as it includes substantial speed
-improvements. See the final section below for details on
-how you can implement your own contexts. When you do not need to include a custom context, a short-hand for specifying
-`HashVersion{2}()` is to call `stable_hash(x; version=2)`.
+Hashing makes use of [`stable_type_name`](https://beacon-biosignals.github.io/StableHashTraits.jl/stable/api/#StableHashTraits.stable_type_name) which is a hash of `parentmodule(T) * string(T)` for type `T`, with a few additional regularizations to ensure e.g. `Core.` values become `Base.` values (as what is in `Core` vs. `Base` changes across julia versions). This function also ensures that no anonymous values (those that include `#`) are hashed, as these are not stable across sessions.
```

Comment:
since `string` is printing, couldn't `string(T)` change? maybe we can use `nameof`?

Reply:
Oh, my mmistake, it is `string(nameof(T))`